### ISSUE TITLE
"when one exists" problem

### ISSUE THREAD

**Issue by yalov** [Ben3eeE/atom-honor-altgraph#1]
There is no `AltGr`+`s` character at my keyboard layouts.
So I created one, for testing purpose, inside keymap.cson:
```
'atom-workspace atom-text-editor ':
  'ctrl-alt-s': 'editor:toggle-line-comments'
```
There is no AltGr character, and there is hotkey, but hotkey does not work.
key binding resolver shows 's' for `AltGr`+`s`

UPD. It's not works (kbr shows 's') for modified English layout, but it works (kbr shows 'ctrl-alt-s') for modified russian layout.  It possible corresponds to https://github.com/atom/atom-keymap/pull/184

**Comment by Ben3eeE** [comment]
This is because of a problem in the keymap that Atom builds for your keyboard layout, the s character (and some others) are 2 characters wide.

![image](https://cloud.githubusercontent.com/assets/1058982/23586207/c27ae33c-0190-11e7-8790-832eacf8f895.png)

**Comment by yalov** [comment]
There isn't any extra char in the source (for MSKLC 1.4) of the eng keyboard layout
https://github.com/yalov/yarbur-keyboard-layouts/blob/master/YarEn200.klc
line 63.

**Comment by Ben3eeE** [comment]
Yeah, there is one issue https://github.com/atom/atom-keymap/issues/188#issuecomment-262910039 where it happens because it maps dead keys incorrectly. But your layout doesn't have any dead keys on the R key so not sure why the keymap is generated this way.

**Comment by Ben3eeE** [comment]
Turns out it actually is caused by dead keys in your layout. I opened an upstream PR to fix (add a workaround for) the keymap generation for these layouts https://github.com/atom/keyboard-layout/pull/29.

**Comment by Ben3eeE** [comment]
https://github.com/atom/keyboard-layout/pull/29 is now merged. This will be fixed starting from Atom 1.18. I tested with your Layout and for modified English layout it resolves to `ctrl-alt-s`.
